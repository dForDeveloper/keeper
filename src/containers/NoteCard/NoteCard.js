import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import uncheckedicon from '../../images/uncheckedicon.svg';
import checkedicon from '../../images/checkedicon.svg';

export class NoteCard extends Component {
  render() {
    const { id, title, listItems } = this.props;
    const incompleteItems = listItems.filter(item => !item.isComplete);
    const completeItems = listItems.filter(item => item.isComplete);
    const result = [];
    result.push(incompleteItems.map(item => {
      const { description, id } = item;
      return (
        <span key={id} className='NoteForm--span--incomplete'>
          <img
            src={uncheckedicon}
            className='NoteForm--icon--unchecked'
            alt='unchecked icon'
          />
          <p className='NoteForm--p--incomplete'>{description}</p>
        </span>
      );
    }));
    result.push(completeItems.map(item => {
      const { description, id } = item;
      return (
        <span key={id} className='NoteForm--span--complete'>
          <img
            src={checkedicon}
            className='NoteForm--icon--checked'
            alt='checked icon'
          />
          <p className='NoteForm--p--complete'>{description}</p>
        </span>
      );
    }));

    return (
      <Link to={'/notes/' + id} className='NoteCard'>
        <h3>{title}</h3>
        <div>{result}</div>
      </Link>
    )
  }
}

export default NoteCard;

NoteCard.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string,
  listItems: PropTypes.array
}