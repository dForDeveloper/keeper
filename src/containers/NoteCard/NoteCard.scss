.NoteCard {
  display: inline-block;
  width: 240px;
  padding: 16px 8px;
  border-radius: 10px;
  text-decoration: none;
  color: #000;
  margin: 8px;
  box-shadow: $light-shadow;
  
  &:hover {
    box-shadow: $heavy-shadow;
  }
}

.NoteCard--h3 {
  font-weight: 400;
  font-size: 1.1rem;
  padding-left: 13px;
  text-align: left;
  overflow-wrap: break-word;
}

.NoteCard--span--incomplete,
.NoteCard--span--complete {
  @extend .NoteForm--span--incomplete;
  font-weight: 300;
  font-size: 1rem;
  margin: 8px;
  word-break: break-all;
}

.NoteCard--ellipsis {
  text-align: left;
  margin: 8px;
  padding-left: 8px;
}