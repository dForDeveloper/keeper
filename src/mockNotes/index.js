export const mockNotes = [
  { 
    id: 'ijf', 
    title: 'Learn Redux', 
    listItems: [
      { 
        id: 'asg', 
        description: 'read through lesson', 
        isComplete: false
      }, 
      { 
        id: 'rhf', 
        description: 'follow along with tutorial', 
        isComplete: false
      }
    ]
  },
  { 
    id: 'pdo', 
    title: 'Start portfolio', 
    listItems: [
      { 
        id: 'cat', 
        description: 'Setup user account', 
        isComplete: false
      }
    ]
  },
  { 
    id: 'bah', 
    title: 'Grocery shopping', 
    listItems: [
      { 
        id: 'rey', 
        description: 'Stop by King Soopers', 
        isComplete: false
      }, 
      { 
        id: 'moo', 
        description: 'Stop by Sprouts', 
        isComplete: false
      }
    ]
  },
  { 
    id: 'fah', 
    title: 'Make dinner', 
    listItems: [
      { 
        id: 'ley', 
        description: 'gather ingredients', 
        isComplete: false
      }, 
      { 
        id: 'mei', 
        description: 'preheat oven', 
        isComplete: false
      }
    ]
  }
]

export const mockNote = {
  id: 'rzz',
  title: 'Workout',
  listItems: [
    {
      id: 'lpo',
      description: '20 situps',
      isComplete: false
    },
    {
      id: 'lqw',
      description: '15 pushups',
      isComplete: false
    }
  ]
}