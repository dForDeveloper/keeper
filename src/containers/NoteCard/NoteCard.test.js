import React from 'react';
import { shallow } from 'enzyme';
import { NoteCard } from './NoteCard';
import { mockNotes, mockNote, mockNoteLong } from '../../mockNotes';

describe('NoteCard', () => {
  let wrapper;
  const mockProps = {
    ...mockNote,
    index: 0,
    notes: mockNotes,
    putAllNotes: jest.fn()
  }
  const mockEvent = {
    dataTransfer: {
      setData: jest.fn(),
      getData: jest.fn(() => 1)
    },
    preventDefault: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(<NoteCard {...mockProps} />);
  });

  it('should pass snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should pass snapshot test when there are more than 10 items', () => {
    wrapper = shallow(<NoteCard {...mockNoteLong} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleDrag', () => {
    it('should call event.dataTransfer.setData with the correct params', () => {
      wrapper.find('.NoteCard--background-lavender').simulate('dragstart', mockEvent);
      expect(mockEvent.dataTransfer.setData).toHaveBeenCalledWith('draggedIndex', 0);
    });
  });
  
  describe('handleDragOver', () => {
    it('should call event.preventDefault()', () => {
      wrapper.find('.NoteCard--background-lavender').simulate('dragover', mockEvent);
      expect(mockEvent.preventDefault).toHaveBeenCalled();
    });
  });

  describe('handleDrop', () => {
    it('should call putAllNotes with the updated array of notes', () => {
      wrapper.find('.NoteCard--background-lavender').simulate('drop', mockEvent);
      const expected = [mockNotes[1], mockNotes[0], ...mockNotes.slice(2)];
      expect(mockProps.putAllNotes).toHaveBeenCalledWith(expected);
    });
  });
});