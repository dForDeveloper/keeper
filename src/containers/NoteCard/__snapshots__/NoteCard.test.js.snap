// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteCard should pass snapshot test 1`] = `
<Link
  replace={false}
  to="/notes/rzz"
>
  <div
    className="NoteCard--background-lavender"
    draggable={true}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
  >
    <h3
      className="NoteCard--h3"
    >
      Workout
    </h3>
    <div>
      <span
        className="NoteCard--span--incomplete"
        key="lpo"
      >
        <img
          alt="unchecked icon"
          className="NoteForm--icon--unchecked"
          src="uncheckedicon.svg"
        />
        <p
          className="NoteForm--p--incomplete"
        >
          20 situps
        </p>
      </span>
      <span
        className="NoteCard--span--complete"
        key="lqw"
      >
        <img
          alt="checked icon"
          className="NoteForm--icon--checked"
          src="checkedicon.svg"
        />
        <p
          className="NoteForm--p--complete"
        >
          15 pushups
        </p>
      </span>
    </div>
    <span
      className="NoteCard--icon-container"
    >
      <img
        alt="edit icon"
        className="NoteCard--delete-icon"
        src="edit.svg"
      />
      <img
        alt="delete icon"
        className="NoteCard--edit-icon"
        src="delete.svg"
      />
    </span>
  </div>
</Link>
`;

exports[`NoteCard should pass snapshot test when there are more than 10 items 1`] = `
<Link
  replace={false}
  to="/notes/bah"
>
  <div
    className="NoteCard--background-undefined"
    draggable={true}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
  >
    <h3
      className="NoteCard--h3"
    >
      Grocery shopping
    </h3>
    <div>
      <span
        className="NoteCard--span--incomplete"
        key="rey"
      >
        <img
          alt="unchecked icon"
          className="NoteForm--icon--unchecked"
          src="uncheckedicon.svg"
        />
        <p
          className="NoteForm--p--incomplete"
        >
          Stop by King Soopers
        </p>
      </span>
      <span
        className="NoteCard--span--incomplete"
        key="fde"
      >
        <img
          alt="unchecked icon"
          className="NoteForm--icon--unchecked"
          src="uncheckedicon.svg"
        />
        <p
          className="NoteForm--p--incomplete"
        >
          Stop by King Soopers
        </p>
      </span>
      <span
        className="NoteCard--span--incomplete"
        key="dfg"
      >
        <img
          alt="unchecked icon"
          className="NoteForm--icon--unchecked"
          src="uncheckedicon.svg"
        />
        <p
          className="NoteForm--p--incomplete"
        >
          Stop by King Soopers
        </p>
      </span>
      <span
        className="NoteCard--span--incomplete"
        key="ert"
      >
        <img
          alt="unchecked icon"
          className="NoteForm--icon--unchecked"
          src="uncheckedicon.svg"
        />
        <p
          className="NoteForm--p--incomplete"
        >
          Stop by King Soopers
        </p>
      </span>
      <span
        className="NoteCard--span--incomplete"
        key="tuy"
      >
        <img
          alt="unchecked icon"
          className="NoteForm--icon--unchecked"
          src="uncheckedicon.svg"
        />
        <p
          className="NoteForm--p--incomplete"
        >
          Stop by King Soopers
        </p>
      </span>
      <span
        className="NoteCard--span--incomplete"
        key="jkh"
      >
        <img
          alt="unchecked icon"
          className="NoteForm--icon--unchecked"
          src="uncheckedicon.svg"
        />
        <p
          className="NoteForm--p--incomplete"
        >
          Stop by King Soopers
        </p>
      </span>
      <span
        className="NoteCard--span--incomplete"
        key="rew"
      >
        <img
          alt="unchecked icon"
          className="NoteForm--icon--unchecked"
          src="uncheckedicon.svg"
        />
        <p
          className="NoteForm--p--incomplete"
        >
          Stop by King Soopers
        </p>
      </span>
      <span
        className="NoteCard--span--incomplete"
        key="lkj"
      >
        <img
          alt="unchecked icon"
          className="NoteForm--icon--unchecked"
          src="uncheckedicon.svg"
        />
        <p
          className="NoteForm--p--incomplete"
        >
          Stop by King Soopers
        </p>
      </span>
      <span
        className="NoteCard--span--incomplete"
        key="uip"
      >
        <img
          alt="unchecked icon"
          className="NoteForm--icon--unchecked"
          src="uncheckedicon.svg"
        />
        <p
          className="NoteForm--p--incomplete"
        >
          Stop by King Soopers
        </p>
      </span>
      <span
        className="NoteCard--span--incomplete"
        key="xcz"
      >
        <img
          alt="unchecked icon"
          className="NoteForm--icon--unchecked"
          src="uncheckedicon.svg"
        />
        <p
          className="NoteForm--p--incomplete"
        >
          Stop by King Soopers
        </p>
      </span>
      <p
        className="NoteCard--ellipsis"
        key="..."
      >
        ...
      </p>
    </div>
    <span
      className="NoteCard--icon-container"
    >
      <img
        alt="edit icon"
        className="NoteCard--delete-icon"
        src="edit.svg"
      />
      <img
        alt="delete icon"
        className="NoteCard--edit-icon"
        src="delete.svg"
      />
    </span>
  </div>
</Link>
`;
